packages/wallet/src/wallet.ts

```diff
@@ -656,23 +658,41 @@ export class Wallet extends Signer {
       })
     })
 
-    return [{
-      delegateCall: false,
-      revertOnError: true,
-      gasLimit: ethers.constants.Zero,
-      to: this.context.sequenceUtils!,
-      value: ethers.constants.Zero,
-      nonce: nonce,
-      data: sequenceUtilsInterface.encodeFunctionData(sequenceUtilsInterface.getFunction('publishInitialSigners'), 
-        [
-          this.address,
-          ethers.utils.keccak256(message),
-          this.config.signers.length,
-          filteredSignature,
-          indexed
-        ]
-      )
-    }]
+    const contextRequireFreshSigner = this.context.libs?.requireFreshSigner
+    if (requireFreshSigners.length > 0 && contextRequireFreshSigner === undefined) {
+      throw Error('requireFreshSigners missing library')
+    }
+
+    return [
+      ...(requireFreshSigners.map((signer) => ({
```

Comment:
Would it be better for the requireFreshSigner to take an array of addresses instead? Right now we need to do n contract calls instead of 1.

Reply:
We only use it for the torus signer, and we must likely never use it for anything else. Making it an array would increase the calldata size for our happy path.

Reply:
Oh I see, misread, thought we were iterating over all addresses, not just the `referenceSigner`. All good then